modules/tensor_mechanics/include/userobjects/CrackFrontPointsProvider.h

```diff
@@ -26,4 +26,10 @@ class CrackFrontPointsProvider : public ElementUserObject
    */
   virtual const std::vector<Point>
   getCrackFrontPoints(unsigned int /*num_crack_front_points*/) const = 0;
+
+  /** get a set of normal vectors along a crack front from a XFEM GeometricCutUserObject
+   * @return A vector which contains all crack front normals
+   */
+  virtual const std::vector<RealVectorValue>
+  getCrackPlaneNormals(unsigned int /*num_crack_front_points*/) const = 0;
```

Comment:
Alternatively, I tried to define `CrackMeshCut3DUserObject * _cutter` in CrackFrontPointsProvider.h so that I can use MeshCut3DUserObject::getCrackPlaneNormals in CrackFrontPointsProvider.C by writing something like `_cutter->getCrackPlaneNormals`.  I did similar things in the ParisLaw code before and that worked fine.  However, the new change didn't pass compiling.  

So by far, I don't find a better way to do this.  Will give it another try.  But any hints would be appreciated.